### DISCUSSION TITLE
Multi Repository Setup

### DISCUSSION THREAD

**Discussion by tricktron** [markdoc/markdoc#317] [Q&A] [answered]
Hello Markdoc Team,

We are currently using Asciidoc/Antora to build our internal wiki website, which pulls documentation pages from multiple different git repos. We really like this approach because it allows to keep the docs in the same git repo as the code and you can easily refactor/restructure your docs with your code.

On the other hand, I always struggle with getting new developers to adopt Asciidoc. _Why not markdown_ is the question I always have to answer:smiley:.

How would you setup a multi-repo structure with Markdoc?

**Comment by mfix-stripe** [comment, marked as answer]
Seems like there are multiple parts to this question, so I'll try and address them individually as best I can. 

> Why not markdown is the question I always have to answer😃.

This one is easy to answer when using Markdoc, since it's a superset of the markdown spec. You can write plain markdown in a Markdoc document and it will just work, and you can customize the [nodes](https://markdoc.dev/docs/nodes) down the line if need be. 

> How would you setup a multi-repo structure with Markdoc?

Markdoc is meant to be a quite low-level toolchain for manipulating Markdoc documents, which means you should have tons of flexibility and control for how this is done. One idea that comes to mind is to gather up a "manifest" at build time, which is just a JSON blob of all of your articles, with metadata around their titles, URLs, contents, where to find them, etc. Maybe you have a list of repos to fetch from. Maybe you use git subrepos, and you just grep for all the `**/*.mdoc` files. Either way, once you know the list of all the files, then mapping those files to a [React tree](https://markdoc.dev/docs/examples/react) or [HTML document](https://markdoc.dev/docs/examples/html) and serving them through a web serve is straightforward. 

Does that help?

**Reply by tricktron** [reply]
@mfix-stripe Thank you for the detailed answer. It helps.

Just out of curiosity: How did you setup the stripe docs?  Did you use a multi-repository setup or a docs monorepo? What would you recommend based on your experiences?

**Reply by mfix-stripe** [reply]
Docs monorepo! In fact, all the pages that comprise https://stripe.com/docs are located in one folder in a monorepo.
IMO monorepos provide a lot of advantages, but don't work for all scenarios.

**Reply by kryltsov** [reply]
@mfix-stripe May I ask you to share how do you keep docs in sync workflow wise? If all the docs in a monorepo and let's say you changed how API works and sequential diagram shared in internal doc folder needs updating, also in public docs folder a piece needs to be rewritten as new optional parameters were added in API. Who is tasked to make sure internal and public docs are updated as you go and how automated this process is or can be. Thanks

**Reply by mfix-stripe** [reply]
Hey @kryltsov, it's a team effort! If there are small changes, engineers will just edit the docs directly alongside their feature/API changes. If it's a larger project, a technical writer will partner with a product team to write the new documentation alongside the launch. 

Outside of projects, we have a whole org decided to the Stripe Docs, so they are constantly working on improving the docs on an ongoing basis.

**Reply by kryltsov** [reply]
Thanks!
